### PR TITLE
[14.0][MIG] shopfloor

### PR DESCRIPTION
**Pull request by hparfr**
Another try to migrate shopfloor

### CODE REVIEW

**Comment by simahawk** [comment]
@hparfr to ease reviews it would be better to:

1. create a new branch based on v14
2. pick the module from v13
3. make it uninstallable
4. open a new PR
5. merge it in 14.0
6. rebase the PR here

This way it will be much easier.

**Comment by hparfr** [comment]
@simahawk ok I will do that.

I had so many conflicts in the merging and wasted a lot time across all the dependencies and changes. Having a green CI is a already a big achievement \o/

File: oca_dependencies.txt
Comment (simahawk) [inline oca_dependencies.txt:2]: we don't need this

File: shopfloor/actions/move_line_search.py
Comment (sebalix) [inline shopfloor/actions/move_line_search.py]: Not sure it is working? Should be a string IMO
```suggestion
        priority_lines = [x for x in lines if x.picking_id.priority == "1"]
```
https://github.com/odoo/odoo/blob/8e11a0b9683c9593f50ab0c290a1bdf857e5141e/addons/stock/models/stock_move.py#L20

File: shopfloor/services/delivery.py
Comment (sebalix) [inline shopfloor/services/delivery.py:256]: Why this change?

File: shopfloor/services/delivery.py
Comment (sebalix) [inline shopfloor/services/delivery.py:306]: Same question

File: shopfloor/services/location_content_transfer.py
Comment (sebalix) [inline shopfloor/services/location_content_transfer.py:823]: Ok, now Odoo 14.0 doesn't unreserve lines having a `qty_done > 0`: https://github.com/odoo/odoo/blob/8e11a0b9683c9593f50ab0c290a1bdf857e5141e/addons/stock/models/stock_move.py#L704-L713
I'm wondering if this is the right thing to do here, if the `qty_done` has been set this means an operator processed it, resetting it to 0 is probably not what we want :thinking:

File: shopfloor/services/location_content_transfer.py
Comment (sebalix) [inline shopfloor/services/location_content_transfer.py:881]: Same remark, to check later

File: shopfloor/tests/test_menu_counters.py
Comment (sebalix) [inline shopfloor/tests/test_menu_counters.py:13]: I suppose we should have 2 priority lines and 1 priority picking related to the `menu1` here based on the configuration of pickings in `test_menu_base.py` above, link to my previous comment https://github.com/OCA/wms/pull/247/files#r675964910

**Review by sebalix — COMMENTED** [review]
Well done, thanks! Few minor comments, once fixed/checked I think we should merge this PR ASAP to ease the port of others PRs we did over time.

File: shopfloor/services/delivery.py
Comment (hparfr) [reply shopfloor/services/delivery.py:256]: Because of linters :disappointed:

File: shopfloor/services/location_content_transfer.py
Comment (hparfr) [reply shopfloor/services/location_content_transfer.py:823]: So how you want to achieve it ?

Comment (sebalix) [reply shopfloor/services/location_content_transfer.py:823]: I don't know yet, previously with the 13.0 implementation we didn't care about it, so this should not block the merge for now.

File: shopfloor/tests/test_menu_counters.py
Comment (sebalix) [reply shopfloor/tests/test_menu_counters.py:13]: So the result is OK here: `picking1` and `picking3` are assigned but has a low priority (set to `"0"`), while the picking2 which is as urgent is still `confirmed` so it has no line counted here.

**Review by sebalix — APPROVED** [review]

**Comment by sebalix** [comment]
@simahawk if it's OK for you we could merge this one

**Review by simahawk — APPROVED** [review]

**Pull request merged**
